e2etest/pkg/frr/config/config.go

```diff
@@ -105,3 +103,34 @@ func SetBGPConfig(testDirName string, config string) error {
 
 	return nil
 }
+
+// Set daemons config file.
+func SetDaemonsConfig(testDirName string, rc RouterConfig) error {
+	path := fmt.Sprintf("%s/%s", testDirName, consts.DaemonsConfigFile)
+	tpl, err := template.ParseFiles(path)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to parse %s", path)
+	}
+
+	f, err := os.Create(path)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to open file %s", path)
+	}
+
+	defer f.Close()
+
+	type Template struct {
+		BGPPort uint16
+	}
+
+	info := Template{
+		BGPPort: rc.BGPPort,
+	}
+
+	err = tpl.Execute(f, info)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to update %s", path)
+	}
+
+	return nil
+}
```

Comment:
why we have this function here and its used only by test ?

Reply:
what do you mean?

Reply:
just curious why you add it here not in the testfile itself